### PR TITLE
Additional changes for Layer State typedef compatibility

### PR DESCRIPTION
**Pull request by drashna**
## Description

This updates the documents, and the tri-layer functions to use the correct code. 

Additionally, it adds `get_highest_layer(state)` (subject to renaming) to replace the `biton32` function with the correct version of the function. 
 
## Types of Changes
- [x] Core
- [x] Enhancement/optimization
- [x] Documentation

## Checklist
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: tmk_core/common/eeconfig.c [hunk 1]
```diff
@@ -8,7 +8,7 @@
 #include "eeprom_stm32.h"
 #endif
 
-extern uint32_t default_layer_state;
+extern uint32_t  default_layer_state;
```
Comment (fauxpark) [inline tmk_core/common/eeconfig.c]:
```suggestion
extern uint32_t default_layer_state;
```

:P

File: quantum/quantum.h [hunk 2]
```diff
@@ -224,7 +224,7 @@ void send_char(char ascii_code);
 
 // For tri-layer
 void update_tri_layer(uint8_t layer1, uint8_t layer2, uint8_t layer3);
-uint32_t update_tri_layer_state(uint32_t state, uint8_t layer1, uint8_t layer2, uint8_t layer3);
+layer_state_t  update_tri_layer_state(layer_state_t  state, uint8_t layer1, uint8_t layer2, uint8_t layer3);
```
Comment (fauxpark) [inline quantum/quantum.h]:
```suggestion
layer_state_t update_tri_layer_state(layer_state_t state, uint8_t layer1, uint8_t layer2, uint8_t layer3);
```

File: quantum/quantum.c [hunk 3]
```diff
@@ -866,9 +866,9 @@ void set_single_persistent_default_layer(uint8_t default_layer) {
   default_layer_set(1U<<default_layer);
 }
 
-uint32_t update_tri_layer_state(uint32_t state, uint8_t layer1, uint8_t layer2, uint8_t layer3) {
-  uint32_t mask12 = (1UL << layer1) | (1UL << layer2);
-  uint32_t mask3 = 1UL << layer3;
+layer_state_t  update_tri_layer_state(layer_state_t  state, uint8_t layer1, uint8_t layer2, uint8_t layer3) {
```
Comment (fauxpark) [inline quantum/quantum.c]:
```suggestion
layer_state_t update_tri_layer_state(layer_state_t state, uint8_t layer1, uint8_t layer2, uint8_t layer3) {
```

File: quantum/quantum.c [hunk 4]
```diff
@@ -866,9 +866,9 @@ void set_single_persistent_default_layer(uint8_t default_layer) {
...
+  layer_state_t  mask12 = (1UL << layer1) | (1UL << layer2);
```
Comment (fauxpark) [inline quantum/quantum.c]:
```suggestion
  layer_state_t mask12 = (1UL << layer1) | (1UL << layer2);
```

File: quantum/quantum.c [hunk 5]
```diff
@@ -866,9 +866,9 @@ void set_single_persistent_default_layer(uint8_t default_layer) {
...
+  layer_state_t  mask3 = 1UL << layer3;
```
Comment (fauxpark) [inline quantum/quantum.c]:
```suggestion
  layer_state_t mask3 = 1UL << layer3;
```

File: tmk_core/common/eeconfig.c (hunk 1 above)
Comment (drashna) [reply tmk_core/common/eeconfig.c]: Well, really: 
```suggestion
extern layer_state_t default_layer_state;
```

**Pull request merged**